packages/block-library/src/paragraph/use-enter.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef } from '@wordpress/element';
+import { useRefEffect } from '@wordpress/compose';
+import { ENTER } from '@wordpress/keycodes';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { hasBlockSupport } from '@wordpress/blocks';
+
+export function useExitOnEnterAtEnd( props ) {
+	const { moveBlocksToPosition } = useDispatch( blockEditorStore );
+	const {
+		getBlockRootClientId,
+		getBlockIndex,
+		getBlockOrder,
+		getBlockName,
+	} = useSelect( blockEditorStore );
+	const propsRef = useRef( props );
+	propsRef.current = props;
```

Comment:
Ideally we should make a `useFreshRef` utility hook in compose.

Reply:
I wonder if this should be wrapped in a useEffect, it seems weird to assign current in render

Reply:
If I'm not mistaken this looks like a ["Latest Ref Pattern"](https://epicreact.dev/the-latest-ref-pattern-in-react/):

```js
function useLatestRef(value) {
  const ref = useRef(value);
  useLayoutEffect(() => {
    ref.current = value;
  });
  return ref;
}
```